src/server/services/wmts/qgswmtsgetcapabilities.cpp

```diff
@@ -412,15 +412,33 @@ namespace QgsWmts
             precision = 6;
           }
 
+          bool invert = crs.hasAxisInverted();
+
           QDomElement bboxElement = doc.createElement( QStringLiteral( "ows:BoundingBox" ) );
           bboxElement.setAttribute( QStringLiteral( "crs" ), tms.ref );
           QDomElement lowerCornerElement = doc.createElement( QStringLiteral( "ows:LowerCorner" ) );
-          QDomText lowerCornerText = doc.createTextNode( qgsDoubleToString( QgsServerProjectUtils::floorWithPrecision( rect.xMinimum(), precision ), precision ) + ' ' + qgsDoubleToString( QgsServerProjectUtils::floorWithPrecision( rect.yMinimum(), precision ), precision ) );
-          lowerCornerElement.appendChild( lowerCornerText );
+          if ( invert )
```

Comment:
this could be written in a more compact way by having firstCoord = rect.xMinimum() and secondCoord = rect.yMinimum() variables and using std::swap() in the invert case

Reply:
Yes indeed, thanks :+1: